web_app/tests/test_positions.py

```diff
@@ -0,0 +1,446 @@
+            "invalid_supply_token",
+            "valid_wallet_id",
+            {
+                "supply_token": "mock_supply_token",
+                "debt_token": "mock_debt_token",
+                "pool_key": {
+                    "token0": "mock_token0",
+                    "token1": "mock_token1",
+                    "fee": "mock_fee",
+                    "tick_spacing": "mock_tick_spacing",
+                    "extension": "mock_extension",
+                },
+                "supply_price": 0,
+                "debt_price": 0,
+            },
+        ),
+        (
+            "valid_supply_token",
+            "invalid_wallet_id",
+            {
+                "supply_token": "mock_supply_token",
+                "debt_token": "mock_debt_token",
+                "pool_key": {
+                    "token0": "mock_token0",
+                    "token1": "mock_token1",
+                    "fee": "mock_fee",
+                    "tick_spacing": "mock_tick_spacing",
+                    "extension": "mock_extension",
+                },
+                "supply_price": 0,
+                "debt_price": 0,
+            },
+        ),
+    ],
+)
+@pytest.mark.anyio
+async def test_get_repay_data_success(
+    client: AsyncClient, supply_token, wallet_id, mock_repay_data
+) -> None:
+    """
+    Test for successfully retrieving repayment data for 
+    different combinations of wallet ID and supply token.
+
+    Args:
+        client (AsyncClient): The test client for the FastAPI application.
+        supply_token (str): The token used for supply.
+        wallet_id (str): The wallet ID of the user.
+        mock_repay_data (dict): Mocked repayment data.
+
+    Returns:
+        None
+    """
+    with (
+        patch(
+            "web_app.contract_tools.mixins.deposit.DepositMixin.get_repay_data"
+        ) as mock_get_repay_data,
+        patch(
+            "web_app.db.crud.PositionDBConnector.get_contract_address_by_wallet_id"
+        ) as mock_get_contract_address,
+        patch(
+            "web_app.db.crud.PositionDBConnector.get_position_id_by_wallet_id"
+        ) as mock_get_position_id,
+    ):
+        mock_get_repay_data.return_value = mock_repay_data
+        mock_get_contract_address.return_value = "mock_contract_address"
+        mock_get_position_id.return_value = 123
+        mock_get_repay_data.return_value = mock_repay_data
+        DepositMixin.get_repay_data = mock_get_repay_data
+        response = client.get(
+            f"/api/get-repay-data?supply_token={supply_token}&wallet_id={wallet_id}"
+        )
+        mock_get_contract_address.assert_called_once_with(wallet_id)
+        mock_get_position_id.assert_called_once_with(wallet_id)
+        mock_get_repay_data.assert_called_once_with(supply_token)
+        expected_response = {
+            **mock_repay_data,
+            "contract_address": "mock_contract_address",
+            "position_id": "123",
+        }
+        assert response.ok
```

Comment:
Please, change it to `response.is_success` ,this object, doesn't have `ok` method, sorry